UX: Show remove password button without suspense

Drenmi opened the pull request:
### What is this change?

We have this slightly awkward UX on the <kbd>Preferences</kbd> > <kbd>Security</kbd> page, where we offer to remove the password from an account if it has other means of signing in. You need to first click a cryptic <kbd>Show</kbd> button which may or may not lead you to the button you want. This button loads the associated accounts so we can evaluate whether to show the <kbd>Remove password</kbd> button or not.

This PR loads that data when entering the route, so that we can just show the button up-front. If removing password isn't possible, the button is disabled and shows an explainer underneath.

**Before:**

![password-show-noop](https://github.com/user-attachments/assets/81857495-548a-4942-911d-248f59263ace)

**After:**

<img width="446" height="167" alt="Screenshot 2025-08-29 at 4 01 42 PM" src="https://github.com/user-attachments/assets/b52049d4-30b9-4af8-b819-50f9e798b595" />

Drenmi commented:
It seems the QUnit test is flaky. Will address that.

pmusaraj commented:
@nullchristo can you review this since you worked on this part of the UI recently? Thanks!

Drenmi commented:
> 1. I originally implemented this as a button but there was some feedback on dev from a few people saying they didn't like it being a button, so I made it a link. See /t/131184/12 on dev.

Seems to be mainly a concern about styling. I think it's probably semantically incorrect to make it a link, since it opens a modal rather than navigate, and should have the ability to be disabled. I added the `btn-transparent` class, which has a similar effect on the style, but it also preserves alignment.

> 2. Are you running into the "Show" button even as a normal user editing your own account?
My understanding here was that the purpose of displaying the "Show" button, and not automatically running checkEmail() on the user model, was for handling when admins are editing another user's preferences. In that case, we don't show things which reference the user's email address or associated accounts until the admin explicitly clicks one of the "Show" buttons. I assumed this was a privacy feature.. In that case, won't automatically running checkEmail() in afterModel() circumvent that?

I moved this into a flag that's loaded as part of `staff-info.json` so that we don't accidentally reveal an e-mail unless requested.

nullchristo reviewed (approved):
Looks good!
